src/Compilers/Core/MSBuildTask/InteractiveCompiler.cs

```diff
@@ -190,20 +191,20 @@ private DotnetHost DotnetHostInfo
                     AddCommandLineCommands(commandLineBuilder);
                     var commandLine = commandLineBuilder.ToString();
 
-                    if (string.IsNullOrEmpty(ToolPath))
+                    // ToolExe delegates back to ToolName if the override is not
+                    // set.  So, if ToolExe == ToolName, we know ToolExe is not
+                    // explicitly overriden.  So, if both ToolPath is unset and
+                    // ToolExe == ToolName, we know nothing is overridden, and
+                    // we can use our own csc.
+                    if (string.IsNullOrEmpty(ToolPath) && ToolExe == ToolName)
```

Comment:
This will do a case sensitive comparison of the tool name. Is that a deliberate decision? 

This an area that we are fairly inconsistent on. In general though I think we use case insesitivity more though.

Reply:
Yes. This is checking for the literal exact same string - due to crazyness in `ToolTask`, you can't just check if the user specified an override for `ToolExe`, as it routes to calling `ToolName` if unset. So we check for unset-ness by doing a literal exact comparison to what `ToolName` is. (See comment)

Also note I considered the case where the user specifies `ToolExe` to be `ToolName` exactly - but in that case, we can still treat it as unset, because we can still assert properties about it (since it's the one we're providing).

(Case insensitive comparison would be inaccurate - say, for example, the user specifies `ToolExe` as the same thing as `ToolName` differing only in case. We can then no longer make the assumption that they're identical, because on case-sensitive filesystems (linux), they may be different utilities)